Tests/iaas/entropy/entropy-check.py

```diff
@@ -1,225 +1,162 @@
 
-Return codes:
-0:    Enough entropy is available, no errors or warnings during execution
-
-1:    The cloud parameter or OS_CLOUD env variable wasn't provided
-2:    No connection to the OpenStack Cloud possible
-
-10:   At least one image of a VM couldn't be found
-11:   At least one VM doesn't have the required fixed entropy available
-12:   At least one VM had too many FIPS 140-2 failures during the test
-19:   Error during the VM requirements check routine
-
-20:   At least one image is missing the recommended HWRNG attributes
-29:   Error during the image attribute check routine
-
-30:   At least one flavor is missing the recommended HWRNG attributes
-31:   At least one flavor doesn't have the optionally recommended HWRNG attributes
-39:   Error during the flavor attribute check routine
-
-40:   At least one VM doesn't provide the recommended RNG daemon
-41:   At least one VM doesn't provide the recommended RNG device
-49:   Error during the VM service check routine
-
-All return codes between (and including) 1-19 as well as all return codes ending on 9
-can be seen as failures.
-
 Check given cloud for conformance with SCS standard regarding
 entropy, to be found under /Standards/scs-0101-v1-entropy.md
+
+Return code is 0 precisely when it could be verified that the standard is satisfied.
+Otherwise the return code is the number of errors that occurred; for further information,
+see the log messages on various channels:
+    CRITICAL  for problems preventing the test to complete,
+    ERROR     for violations of requirements,
+    INFO      for violations of recommendations,
+    DEBUG     for background information and problems that don't hinder the test.
 """
-import errno
+from collections import Counter
 import getopt
+import logging
 import os
 import re
 import sys
+import tempfile
 import time
+import warnings
 
 import fabric
-import keystoneauth1.exceptions
 import openstack
 import openstack.cloud
-import invoke
 
 
-server_name = "test"
-security_group_name = "test-group"
-keypair_name = "test-keypair"
+logger = logging.getLogger(__name__)
 
-mand_images = ["Ubuntu 22.04", "Ubuntu 20.04", "Debian 11"]
-rec1_images = ["CentOS 8", "Rocky 8", "AlmaLinux 8", "Debian 10", "Fedora 36"]
-rec2_images = ["SLES 15SP4", "RHEL 9", "RHEL 8"]
-# , "Windows Server 2022", "Windows Server 2019"] we can't test Windows in this case
-sugg_images = ["openSUSE Leap 15.4", "Cirros 0.5.2", "Alpine", "Arch"]
+# use telling names here, maybe even reuse the security group in case of a leak?
+SERVER_NAME = "scs-0101-server"
+SECURITY_GROUP_NAME = "scs-0101-group"
+KEYPAIR_NAME = "scs-0101-keypair"
+
+IMAGE_ATTRIBUTES = {
+    # https://docs.openstack.org/glance/latest/admin/useful-image-properties.html#image-property-keys-and-values
+    # type: str
+    "hw_rng_model": "virtio",
+}
+FLAVOR_ATTRIBUTES = {
+    # https://docs.openstack.org/nova/latest/configuration/extra-specs.html#hw-rng
+    # type: bool
+    "hw_rng:allowed": True,
+}
```

Comment:
These attributes are optional. Not sure what to say when they are missing...

Reply:
I know what you mean, but to be fair, the HW attributes are also kind of optional. I would just handle them like the other attributes and make them optional, since they all just produce info level logs.

Reply:
Looks and sounds good.